projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/controller/DataJobsSecretsController.java

```diff
@@ -23,19 +23,61 @@
 public class DataJobsSecretsController implements DataJobsSecretsApi {
   static Logger log = LoggerFactory.getLogger(DataJobsSecretsController.class);
 
+  //  private final FeatureFlags featureFlags;
+
+  //  private final JobSecretsService secretsService;
+
+  //  @Autowired
+  //  public DataJobsSecretsController(FeatureFlags featureFlags, JobSecretsService secretsService)
+  // {
+  //  public DataJobsSecretsController(FeatureFlags featureFlags, JobSecretsService secretsService)
+  // {
+  //    this.featureFlags = featureFlags;
+  //    this.secretsService = secretsService;
+  //  }
+
   @Override
   public ResponseEntity<Void> dataJobSecretsUpdate(
       String teamName, String jobName, String deploymentId, Map<String, Object> requestBody) {
     log.debug("Updating secrets for job: {}", jobName);
+
+    //    TODO: Remove after adding tests
     throw new ResponseStatusException(
         HttpStatus.NOT_IMPLEMENTED, "Secrets service is not implemented");
+
+    //    TODO: Working implementation. Uncomment after adding tests
+    //    if (featureFlags.isVaultIntegrationEnabled()) {
+    //      secretsService.updateJobSecrets(jobName, requestBody);
+    //      return ResponseEntity.noContent().build();
+    //    }
+    //
+    //    throw new ResponseStatusException(
```

Comment:
Use the format of the rest of the exceptions - e.g. DataJobDeploymentNotFoundException.

Reply:
fixed.